packages/connections/src/XHRConnection.js

```diff
@@ -0,0 +1,170 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+
+export const ALLOWED_METHODS = ["GET", "POST", "PUT", "DELETE"];
+
+/**
+ * Basic XHR Connection
+ * @todo add/remove headers
+ */
+export default class XHRConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of XHRConnection
+   * @constructor
+   * @param {string} url – URL to be fetched
+   * @param {object} [options]
+   * @param {string} [options.method=GET] – used method
+   * @param {*} [options.body] – payload for request
+   * @param {object} [options.headers] – additional headers (like content-type)
+   * @param {string} [options.responseType]
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    const {
+      method = "GET",
+      body = null,
+      headers = {},
+      responseType = "json"
+    } = options;
+
+    if (ALLOWED_METHODS.includes(method) === -1) {
+      throw new Error("Invalid method for XHR Request.");
+    }
+
+    /**
+     * @property {string}
+     * @protected
+     * @name XHRConnection#method
+     */
+    Object.defineProperty(this, "method", { value: method });
+
+    /**
+     * @property {*}
+     * @protected
+     * @name XHRConnection#body
+     */
+    Object.defineProperty(this, "body", { value: body });
+
+    /**
+     * @property {object}
+     * @protected
+     * @name XHRConnection#headers
+     */
+    Object.defineProperty(this, "headers", { value: headers });
+
+    /**
+     * @property {string}
+     * @protected
+     * @name XHRConnection#responseType
+     */
+    Object.defineProperty(this, "responseType", { value: responseType });
```

Comment:
Let's make this an ENUM similar to `method` and list the ENUMs in the error for a better debug experience.

Reply:
I'm reluctant to add these as some browsers allow types like `moz-blob` or `ms-stream` and I wouldn't want to maintain a list with all possible values. I'm happy to add a check if we restrict the available types and only allow the following types:

Value | Data type of response property
-- | --
"arraybuffer" | ArrayBuffer
"blob" | Blob
"document" | Document
"json" | JavaScript object, parsed from a JSON string returned by the server
"text" | DOMString 

Reply:
I think it's better to be explicit and choose our support. We shouldn't be allowing browser specific response values.

Reply:
✔️  Done